src/awkward/_v2/contents/listoffsetarray.py

```diff
@@ -570,6 +570,220 @@ def _localindex(self, axis, depth):
                 self._parameters,
             )
 
+    def _is_unique(self, negaxis, starts, parents, outlength):
+        if len(self._offsets) - 1 == 0:
+            return True
+
+        branch, depth = self.branch_depth
+
+        if (
+            self.parameter("__array__") == "string"
+            or self.parameter("__array__") == "bytestring"
+        ):
+            if branch or (negaxis is not None and negaxis != depth):
+                raise ValueError(
+                    "array with strings can only be checked on uniqueness with axis=-1"
+                )
+
+            # FIXME: check validity error
+
+            if isinstance(self._content, ak._v2.contents.NumpyArray):
+                out, outoffsets = self._content._as_unique_strings(self._offsets)
+                out2 = ak._v2.contents.listoffsetarray.ListOffsetArray(
+                    outoffsets,
+                    out,
+                    None,
+                    self._parameters,
+                )
+                return len(out2) == len(self)
+
+        if negaxis is None:
+            return self._content._is_unique(negaxis, starts, parents, outlength)
+
+        if not branch and (negaxis == depth):
+            return self._content._is_unique(negaxis - 1, starts, parents, outlength)
+        else:
+            nplike = self.nplike
+
+            nextparents = ak._v2.index.Index64.empty(
+                self._offsets[-1] - self._offsets[0], nplike
+            )
+
+            self._handle_error(
+                nplike[
+                    "awkward_ListOffsetArray_reduce_local_nextparents_64",
+                    nextparents.dtype.type,
+                    self._offsets.dtype.type,
+                ](
+                    nextparents.to(nplike),
+                    self._offsets.to(nplike),
+                    len(self._offsets) - 1,
+                )
+            )
+            starts = self._offsets[:-1]
+
+            return self._content._is_unique(negaxis, starts, nextparents, outlength)
+
+    def _unique(self, negaxis, starts, parents, outlength):
+        nplike = self.nplike
+        if len(self._offsets) - 1 == 0:
+            return self
+
+        branch, depth = self.branch_depth
+
+        if (
+            self.parameter("__array__") == "string"
+            or self.parameter("__array__") == "bytestring"
+        ):
+            if branch or (negaxis != depth):
+                raise ValueError("array with strings can only be sorted with axis=-1")
```

Comment:
I think this also should be a `np.AxisError`.

Reply:
done